app/src/main/java/org/simple/clinic/appconfig/StatesFetcher.kt

```diff
@@ -13,11 +13,16 @@ class StatesFetcher @Inject constructor(
   @Throws(ConnectException::class)
   fun fetchStates(deployment: Deployment): List<State> {
     val statesApi = createStatesApi(deployment)
-    return statesApi
+    val statesPayload = statesApi
         .fetchStates()
+        .execute()
+        .body()
+    val states = statesPayload?.states.orEmpty()
+
+    return states
         .map { state ->
           State(
-              displayName = state,
+              displayName = state.displayName,
```

Comment:
Q: What happens here if we receive an empty statePayload?

Reply:
I don't think that's possible. But in that scenario, this `map` won't be called and we send an empty list and we show an empty recycler view. Since this list is generated by server folks and not user dependent, I don't think we have to handle those cases.